src/index.ts

```diff
@@ -315,7 +354,11 @@ export class Tado {
    * @returns A promise that resolves to an array of Device objects.
    */
   getDevices(home_id: number): Promise<Device[]> {
-    return this.apiCall(`/api/v2/homes/${home_id}/devices`);
+    if (this.isHomeX(home_id)) {
+      return this.apiCallX(`/homes/${home_id}/roomsAndDevices`);
```

Comment:
Tried hitting this end point. I can see see a response, but the data structure is different to `Devices[]`.

```json
{
  "rooms": [],
  "otherDevices": []
}
```

Reply:
I didn't even try to hit it, how ridiculous. I totally should have just tried 😂

I guess that's a decent pointer that these might need to be separate API calls or possibly separate classes.

Reply:
Yeah I think the best way to handle this might be to split into multiple classes, sharing a base class. And we should just assume that someone using the client knows what generation they want to use.

Otherwise, trying to keep everything unified might have a high maintenance cost.

I am also really curious as to why these endpoints are not even mentioned in the mobile apps seeing as line x is already being used within the same app. This leads me to guess v2 api supports line x homes too. Definitely will need to verify that before merging any of this.

Reply:
Agreed. I'm definitely leaning toward two classes. 

If the v2 endpoints still work then I guess the user can instantiate both classes. 

Interesting that it isn't in the app. I wonder if there's some new app coming.

Reply:
The latest push splits TadoX into a separate class. I also renamed all of the TadoX calls to be more accurate to the underlying call - this means no conflicts with the v2 API so `TadoX extends Tado` for now.